chainstate-storage/src/internal/test.rs

```diff
@@ -156,24 +156,20 @@ fn get_set_transactions() {
         let thr1 = {
             let store = Store::clone(&store);
             common::thread::spawn(move || {
-                let _ = store.transaction_rw().run(|tx| {
-                    let v = tx.get_storage_version()?;
-                    tx.set_storage_version(v + 1)?;
-                    storage::commit(())
-                });
+                let mut tx = store.transaction_rw();
```

Comment:
No more scoped transactions?

Reply:
I decided to remove them since it was not really used in the code at all, only in tests, and there is little point on testing something that is not used in the software itself. Me removing it is just recognition of this reality.

My preference would be to have only the scoped transaction interface, since then you can provide stronger guarantees the storage module is used correctly. For example, it guarantees the transactions are properly finalized, i.e. `close` is called to terminate `ro` transactions and either `commit` or `abort` are called for `rw` transactions. I think it also encourages more principled approach to error handling.